Packs/QuestKace/Integrations/QuestKace/QuestKace.py

```diff
@@ -0,0 +1,677 @@
+        elif shaping_fields:
+            return self._http_request("GET", url_suffix=f"/service_desk/tickets?shaping={shaping_fields}",
+                                      headers=headers)
+        else:
+            return self._http_request("GET", url_suffix=f"/service_desk/tickets", headers=headers)
+
+    def create_ticket_request(self, data: str) -> dict:
+        """Create Ticket
+           Returns:
+               Response from API.
+        """
+        headers = {
+            'Accept': 'application/json',
+            'x-dell-csrf-token': self._token,
+            'x-dell-api-version': '5',
+            'Cookie': self._cookie,
+            'Content-Type': 'application/json'
+        }
+        return self._http_request("POST", url_suffix="/service_desk/tickets", headers=headers, data=data)
+
+    def update_ticket_request(self, ticket_id: str, data: str) -> dict:
+        """Update Ticket.
+           Returns:
+               Response from API.
+        """
+        headers = {
+            'Accept': 'application/json',
+            'x-dell-csrf-token': self._token,
+            'x-dell-api-version': '5',
+            'Cookie': self._cookie,
+            'Content-Type': 'application/json'
+        }
+        return self._http_request("POST", url_suffix=f"/service_desk/tickets/{ticket_id}", headers=headers, data=data)
+
+    def delete_ticket_request(self, ticket_id: str) -> dict:
+        """Delete Ticket.
+           Returns:
+               Response from API.
+        """
+        headers = {
+            'Accept': 'application/json',
+            'x-dell-csrf-token': self._token,
+            'x-dell-api-version': '5',
+            'Cookie': self._cookie,
+            'Content-Type': 'application/json'
+        }
+        return self._http_request("DELETE", url_suffix=f"/service_desk/tickets/{ticket_id}", headers=headers)
+
+    def ticket_by_id_request(self, id: int) -> dict:
+        """Specific ticket details by ID.
+            Args:
+                id: id for filtering by it.
+           Returns:
+               Response from API.
+        """
+        headers = {
+            'Accept': 'application/json',
+            'x-dell-csrf-token': self._token,
+            'x-dell-api-version': '5',
+            'Cookie': self._cookie
+        }
+        filter_fields = f"id eq {id}"
+        return self._http_request("GET", url_suffix=f"/service_desk/tickets?filtering={filter_fields}", headers=headers)
+
+
+def test_module(client: Client, args=None) -> Tuple[str, dict, dict]:
+    """Function which checks if there is a connection with the api.
+        Args:
+            client : Integretion client which communicates with the api.
+            args: Users arguments of the command.
+       Returns:
+           human readable, context, raw response of this command.
+    """
+    response_machines = client.machines_list_request()
+    machines = response_machines.get('Machines')
+    client.update_token()
```

Comment:
do you have to update the token in between each request? I thought it is valid for a longer time.

Reply:
I am not sure. for some reason it gets error 500 using python requests. however in postman it works without the need to login again. Could not figure out  why.